docs/dev/core/new-metric-type.md

```diff
@@ -139,66 +139,43 @@ Each metric type is implemented in its own module.
 
 Add a new file named after your metric, e.g. `glean-core/ffi/src/counter.rs`, and declare it in `glean-core/ffi/src/lib.rs` with `mod counter;`.
 
-In the metric type module add a global map for your metric type and define the destructor:
+In the metric type module define your metric type using the `define_metric` macro.
+This allows referencing the metric name and defines the global map as well as some common functions such as the constructor and destructor.
+Simple operations can be also defined in the same macro invocation:
+
 
 ```rust,noplaypen
-lazy_static! {
-    static ref COUNTER_METRICS: ConcurrentHandleMap<CounterMetric> = ConcurrentHandleMap::new();
-}
-define_handle_map_deleter!(COUNTER_METRICS, glean_destroy_counter_metric);
-```
+use crate::{define_metric, handlemap_ext::HandleMapExtension, GLEAN};
 
-Add a function to create new instances of this metric type:
+define_metric!(CounterMetric => COUNTER_METRICS {
+    new           -> glean_new_counter_metric(),
```

Comment:
Not a problem but oh, I only noticed now that the `new` requires trailing `()`

Reply:
Only because it could also take additional parameters.
I _might_ be able to get rid of that, but not sure.

Reply:
No worries, no need to act on this :)